src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationRepository.cs

```diff
@@ -17,21 +16,90 @@ namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
 
-        public async Task<List<Publication>> GetAllPublicationsForTopic(Guid topicId)
+        public async Task<List<Publication>> GetPublicationsForTopic(Guid topicId)
         {
             return await HydratePublicationForPublicationViewModel(_context.Publications)
                 .Where(publication => publication.TopicId == topicId)
                 .ToListAsync();
         }
 
+        public async Task<List<Publication>> GetPublicationListForUser(
+            Guid userId,
+            Guid? topicId = null)
+        {
+            var publicationsGrantedByPublicationRoleQueryable = _context
+                .UserPublicationRoles
+                .AsQueryable()
+                .Where(userPublicationRole => userPublicationRole.UserId == userId &&
+                                              ListOf(PublicationRole.Owner, PublicationRole.ReleaseApprover)
+                                                  .Contains(userPublicationRole.Role));
+
+            if (topicId.HasValue)
+            {
+                publicationsGrantedByPublicationRoleQueryable =
+                    publicationsGrantedByPublicationRoleQueryable.Where(userPublicationRole =>
+                        userPublicationRole.Publication.TopicId == topicId.Value);
+            }
+
+            var publicationsGrantedByPublicationRole = await publicationsGrantedByPublicationRoleQueryable
+                .Select(userPublicationRole => userPublicationRole.Publication)
+                .ToListAsync();
+
+            var publicationIdsGrantedByPublicationRole = publicationsGrantedByPublicationRole
+                .Select(publication => publication.Id)
+                .ToList();
+
+            var releasesGrantedByReleaseRolesQueryable = _context.UserReleaseRoles
+                .Include(userReleaseRole => userReleaseRole.Release.Publication)
+                .Where(userReleaseRole => userReleaseRole.UserId == userId &&
+                                          userReleaseRole.Role != ReleaseRole.PrereleaseViewer);
+
+            if (topicId.HasValue)
+            {
+                releasesGrantedByReleaseRolesQueryable =
+                    releasesGrantedByReleaseRolesQueryable.Where(userReleaseRole =>
+                        userReleaseRole.Release.Publication.TopicId == topicId.Value);
+            }
+
+            var releasesGrantedByReleaseRoles = await releasesGrantedByReleaseRolesQueryable
+                .Select(userReleaseRole => userReleaseRole.Release)
+                .ToListAsync();
+
+            var publications = new List<Publication>();
+
+            // Add publication view models for the Publications granted directly via Publication roles
+            publications.AddRange(await publicationsGrantedByPublicationRole
+                .SelectAsync(async publication =>
+                    // Include all Releases of the Publication unconditionally
+                    await HydratePublicationForPublicationViewModel(_context.Publications)
```

Comment:
I've left a comment pointing out that this is a method of `PublicationService` which doesn't feel right, so need to take a closer look at this refactoring and what the responsibilities of the repository are vs. the service.

Reply:
I'll rename it to `HydratePublication` and move it to `PublicationRepository` - there is one use of the method in `PublicationService`.

Guess arguably as the method is static, you could move it to a PublicationUtils class - I won't resolve this comment in case either of you think that's preferable.